### PR TITLE
FileManager microservice Rust skel

### PR DESCRIPTION
**Pull request by brainstorm**
... and initial draft/incomplete definition of filemanager microservice client API.

### CODE REVIEW

**Review by victorskl — COMMENTED** [review]
Nice. LGTM so far; will review in more once PR is finalised! 👍

File: skel/rust-api/src/filemanager.rs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+use serde::{Deserialize, Serialize};
+use utoipa::{IntoParams, ToSchema};
+
+use tracing::{ info };
+//type Store = Mutex<Vec<FileManager>>;
+// TODO: SQLx is the store backend, through db.rs
+
+/// Item to do.
+#[derive(Serialize, Deserialize, ToSchema, Clone)]
+pub struct FileManager {
```
Comment (reisingerf) [inline skel/rust-api/src/filemanager.rs:11]: Is that the Rust equivalent of a Django Model class? 
Just wondering about the naming... the FileManager is more the microservice itself where the model would be more something like File/Object/FileObject/... ?

Not at all into Rust, so please ignore it that does not apply!

**Comment by reisingerf** [comment]
Great start, thanks!
(expect stupid noob Rust questions though ;-) )

File: skel/rust-api/src/filemanager.rs (hunk 1 above)
Comment (brainstorm) [reply skel/rust-api/src/filemanager.rs:11]: Yes! I was torn between "File" and "Object", so I left it as "FileManager", which is clearly confusing and wrong, which one would you pick from the former two names? I'll change it :)

Comment (brainstorm) [reply skel/rust-api/src/filemanager.rs:11]: No `File` doesn't clash with names, Rust is pretty good with shadowing names without clashes... let's go with file then.

**Comment by victorskl** [comment]
[OrcaBus V1: Design and Implement File Manager](https://trello.com/c/OmhmlAkN/1426-orcabus-v1-design-and-implement-file-manager)

**Review by victorskl — APPROVED** [review]
Rust skel looks good and LGTM! Happy to merge this ahead first as we would need to base OpenAPI doc arrangement from this PR.

File: skel/rust-api/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            (name = "File", description = "File manager API")
+        )
+    )]
+    struct ApiDoc;
+
+    #[derive(Clone)]
+    struct AppState {}
+    struct SecurityAddon;
+
+    impl Modify for SecurityAddon {
```
Comment (mmalenic) [inline skel/rust-api/src/main.rs:48]: Would it be better if these were outside of the main function?

Comment (brainstorm) [reply skel/rust-api/src/main.rs:48]: Thought exactly the same right after merging it, along with putting a few more comments to guide new user's edits :-S

**Review by reisingerf — COMMENTED** [review]
LATM

File: skel/rust-api/src/main.rs (hunk 2 above)
Comment (brainstorm) [reply skel/rust-api/src/main.rs:48]: https://github.com/umccr/orcabus/commit/483226ff149feddfd155e82b3a9823ef60df1752

**Pull request merged**